pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1177,9 +1178,101 @@
       ::
       ?~  session=(~(get by sessions.authentication-state.state) `@uv`u.session-id)
         %.n
-      ::  is this session still valid?
+      ::  does this session have our id, and is it still valid?
+      ::
+      &(?=(%ours -.identity.u.session) (lte now expiry-time.u.session))
+    ::  +start-session: create a new session with %local or %guest identity
+    ::
+    ++  start-session
+      |=  kind=?(%local %guest)
+      ^-  [[session=@uv =identity moves=(list move)] server-state]
+      =;  [key=@uv sid=identity]
+        :-  :+  key  sid
+            ::  if no session existed previously, we must kick off the
+            ::  session expiry timer
+            ::
+            ?^  sessions.authentication-state.state  ~
+            [duct %pass /sessions/expire %b %wait (add now session-timeout)]~
+        =-  state(sessions.authentication-state -)
+        %+  ~(put by sessions.authentication-state.state)  key
+        [sid (add now session-timeout) ~]
+      ::  create a new session with a fake identity
+      ::
+      =/  sik=@uv  new-session-key
+      :-  sik
+      ?:  ?=(%local kind)  [%ours ~]
+      :-  %fake
+      ::  pre-scramble our ship name into its displayed value, and
+      ::  truncate it to be at most moon-length, so that we can overlay
+      ::  it onto the end of a comet name for visual consistency.
+      ::  to prevent escalation, make sure the guest identity isn't ours.
       ::
-      (lte now expiry-time.u.session)
+      |-
+      =;  nom=@p
+        ?.  =(our nom)  nom
+        $(eny (shas %next-name eny))
+      =/  hat  (end 3^8 (fein:ob our))
+      %^  cat  3  hat
+      %+  rsh  [3 (met 3 hat)]
+      (~(raw og (shas %fake-name eny)) 128)
```

Comment:
I think this would be clearer if it concatenated, then truncated. But it works fine as written.

Reply:
If you're looking for a trivial push to update the diff .... `(end 3^8 foo)` is `(end 6 foo)`, and `(end 3^16 foo)` is `(end 7 foo)`.